README.md

```diff
@@ -141,6 +141,87 @@ using (IDbConnection conn = new SnowflakeDbConnection())
 }
 ```
 
+If you are using a different method for authentication, see the examples below:
+
+* Key-pair authentication
+
+  After setting up [key-pair authentication](https://docs.snowflake.com/en/user-guide/key-pair-auth.html), you can specify the
+  private key for authentication in one of the following ways:
+
+  * Specify the file containing an unencrypted private key:
+
+    ```cs
+    using (IDbConnection conn = new SnowflakeDbConnection())
+    {
+        conn.ConnectionString = "account=testaccount;user=testuser;authenticator=snowflake_jwt;private_key_file={pathToThePrivateKeyFile};db=testdb;schema=testschema";
+
+        conn.Open();
+    
+        conn.Close();
+    }
+    ```
+
+    where:
+
+    * `{pathToThePrivateKeyFile}` is the path to the file containing the unencrypted private key.
+
+  * Specify the file containing an encrypted private key:
+
+    ```cs
+    using (IDbConnection conn = new SnowflakeDbConnection())
+    {
+        conn.ConnectionString = "account=testaccount;user=testuser;authenticator=snowflake_jwt;private_key_file={pathToThePrivateKeyFile};private_key_pwd={passwordForDecryptingThePrivateKey};db=testdb;schema=testschema";
+
+        conn.Open();
+    
+        conn.Close();
+    }
+    ```
+
+    where:
+
+    * `{pathToThePrivateKeyFile}` is the path to the file containing the unencrypted private key.
+    * `{passwordForDecryptingThePrivateKey}` is the password for decrypting the private key.
+
+  * Specify an unencrypted private key (read from a file):
+
+    ```cs
+    using (IDbConnection conn = new SnowflakeDbConnection())
+    {
+        string privateKeyContent = File.ReadAllText({pathToThePrivateKeyFile}).Replace("=", "==");
+
+        conn.ConnectionString = String.Format("account=testaccount;authenticator=snowflake_jwt;user=testuser;private_key={0}", privateKeyContent);
```

Comment:
db=testdb;schema=testschema can be added if you want to use the same set of properties for all connection strings.

Reply:
Thanks! I added this. I also made the order of the parameters in the example more consistent.